trace-utils/src/tracer_payload.rs

```diff
@@ -106,14 +105,138 @@ impl TracerPayloadCollection {
     }
 }
 
+pub type TracerPayloadChunkProcessor<'a> = Box<dyn Fn(&mut pb::TraceChunk, usize) + 'a>;
+
+/// Represents the parameters required to collect trace chunks from msgpack data.
+///
+/// This struct encapsulates all the necessary parameters for converting msgpack data into
+/// a `TracerPayloadCollection`. It is designed to work with the `TryInto` trait to facilitate
+/// the conversion process, handling different encoding types and ensuring that all required
+/// data is available for the conversion.
+///
+/// # Fields
+///
+/// * `process_chunk`: A boxed closure that processes each `TraceChunk`, allowing for custom
+/// * `encoding_type`: The encoding type of the trace data, determining how the data should be
+///   deserialized and processed.
+pub struct TracerPayloadParams<'a> {
+    /// A byte slice containing the serialized msgpack data.
+    data: &'a [u8],
+    /// Reference to `TracerHeaderTags` containing metadata for the trace.
+    tracer_header_tags: &'a TracerHeaderTags<'a>,
+    /// A boxed closure that processes each `TraceChunk`, allowing for custom logic to be applied
+    /// during the conversion process.
+    process_chunk: TracerPayloadChunkProcessor<'a>,
```

Comment:
Do we often use a ChunkProcessor ? I not it feels a bit heavy to have to create an empty closure every time we call it maybe it can be an Option or provide an alternative constructor which creates the empty closure.

Reply:
That's a fair point. It currently only works for `v07` and is only used by the mini-agent. For now, I created a `DefaultTraceChunkProcessor` that can be passed so we don't have to implement the trait just to do a no-op. The compiler _should_ optimize away the call to the processor in `collect_trace_chunks` in this situation. 

I'm inclined to leave it as is for now. I think we may want to do some further refactoring to TracerPayloads in the not to distant future and we may want to think of ways to streamline how some of this works.